5. Unsupervised Clustering.ipynb

```diff
@@ -0,0 +1,1088 @@
+    "import numpy as np\n",
+    "from scipy.stats import spearmanr\n",
+    "from sklearn.cluster import Birch\n",
+    "from sklearn.preprocessing import StandardScaler\n",
+    "from sklearn.metrics import calinski_harabasz_score, davies_bouldin_score\n",
+    "from umap import UMAP\n",
+    "from sklearn.decomposition import PCA\n",
+    "import holoviews as hv\n",
+    "import holoviews.operation.datashader as hd\n",
+    "from bokeh.models import HoverTool\n",
+    "from holoviews import opts\n",
+    "import datashader as ds\n",
+    "from matplotlib.colors import ListedColormap\n",
+    "from bokeh.palettes import Viridis256\n",
+    "from bokeh.plotting import show\n",
+    "\n",
+    "hv.extension('bokeh')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import skimage.exposure\n",
+    "import shapely\n",
+    "import shapely.wkt\n",
+    "import math\n",
+    "import shapely.affinity\n",
+    "import matplotlib.pyplot as plt"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Define variables and OMERO IDs that will be used for the next steps"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Define the image and table IDs\n",
+    "# Image ID will be used to link the uploaded table to the image\n",
+    "IMAGE_ID = 1912\n",
+    "# Table ID\n",
+    "TABLE_FILE_ID = 14510\n",
+    "# Analysis ID - distinct name for the local files and the table\n",
+    "# Can also use uuid lib to generate a random ID\n",
+    "ANALYSIS_ID = \"95a5420c-9732-4a49-961c-cf43264a7b50\"\n",
+    "\n",
+    "# Where to store the downloaded table?\n",
+    "table_path = f\"./objects_{ANALYSIS_ID}.csv\"\n",
+    "# The name will be used as OMERO.table name\n",
+    "output_table_name = f\"{ANALYSIS_ID}_unsupervised_clustering\"\n",
+    "# Store outputs locally\n",
+    "output_table_path = f\"{output_table_name}.csv\""
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Load the table from OMERO and save it locally for future us"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "If [omero-user-token](https://github.com/glencoesoftware/omero-user-token) is installed and valid [omero2pandas](https://github.com/glencoesoftware/omero2pandas) will use it automatically for the OMERO.server connection otherwise it will ask for the connection details."
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "##### Without omero-user-token"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Connected to apps.dev.omero-plus.io\n"
```

Comment:
@emilroz Are we okay with exposing the URL of the dev server? It appears in a number of outputs

Reply:
I think that's ok.